### PR TITLE
Run button now appears if re-run is unavailable on job result list an…

### PR DESCRIPTION
**Pull request by briddo**
…d job result detail.
# Closes: #2206 
# What's Changed
The job run button now appears on a job result detail page or the list if this job has no saved input data.

# TODO
- [ ] Unit, Integration Tests
- [ ] Release Note

### CODE REVIEW

File: nautobot/extras/tables.py [hunk 1]
```diff
@@ -705,8 +705,9 @@ class JobResultTable(BaseTable):
                     <i class="mdi mdi-repeat"></i>
                 </a>
             {% else %}
-                <a href="#" class="btn btn-xs btn-default disabled" title="No saved job arguments, cannot be re-run">
-                    <i class="mdi mdi-repeat-off"></i>
+                <a href="{% url 'extras:job_run' slug=record.job_model.slug %}" class="btn btn-primary btn-xs"
```
Comment (glennmatthews) [inline nautobot/extras/tables.py:708]: Need a check here to make sure `record.job_model` is not None

File: nautobot/extras/templates/extras/jobresult.html [hunk 2]
```diff
@@ -33,12 +33,17 @@
...
-        <a href="{% url 'extras:job_run' slug=result.job_model.slug %}?kwargs_from_job_result={{ result.pk }}"
-           class="btn btn-info">
-            <span class="mdi mdi-repeat" aria-hidden="true"></span> Re-Run
-        </a>
+            <a href="{% url 'extras:job_run' slug=result.job_model.slug %}?kwargs_from_job_result={{ result.pk }}"
+               class="btn btn-info">
+                <span class="mdi mdi-repeat" aria-hidden="true"></span> Re-Run
+            </a>
+        {% else %}
+            <a href="{% url 'extras:job_run' slug=result.job_model.slug %}"
```
Comment (glennmatthews) [inline nautobot/extras/templates/extras/jobresult.html:43]: Need `record.job_model is not None` check here as well

File: nautobot/extras/tables.py (hunk 1 above)
Comment (glennmatthews) [inline nautobot/extras/tables.py:708]: We have some inconsistency in button colors here (btn-success, btn-primary) and in the detail view (btn-info). Not sure the reasoning behind this?

File: nautobot/extras/tables.py [hunk 3]
```diff
@@ -704,6 +704,11 @@ class JobResultTable(BaseTable):
                    class="btn btn-xs btn-success" title="Re-run job with same arguments.">
                     <i class="mdi mdi-repeat"></i>
                 </a>
+            {% elif record.job_model is not None %}
```
[the remark this answers is not in the document]
Comment (briddo) [reply nautobot/extras/tables.py]: Good spot!

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**